rust/chains/hyperlane-cosmos/src/merkle_tree_hook.rs

```diff
@@ -317,6 +317,7 @@ impl Indexer<MerkleTreeInsertion> for CosmosMerkleTreeHookIndexer {
                 }
             })
             .flatten()
+            .map(|(log, meta)| (Indexed::new(log).with_sequence(log.index() as _), meta))
```

Comment:
nit: iiuc for consistency you can use the `impl From<MerkleTreeInsertion> for Indexed<MerkleTreeInsertion>` like we do elsewhere